sampling/rate_limited/test/opentelemetry/sampling/rate_limited/sampler_test.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require('test_helper')
+
+describe(OpenTelemetry::Sampling::RateLimited::Sampler) do
+  describe('#should_sample?') do
+    it('returns DROP if credits_per_second is 0') do
+      sampler = OpenTelemetry::Sampling::RateLimited::Sampler.new(0)
+
+      _(
+        sampler.should_sample?(
+          trace_id: SecureRandom.uuid.to_s,
+          parent_context: nil,
+          links: [],
+          name: SecureRandom.uuid.to_s,
+          kind: :internal,
+          attributes: {}
+        ).sampled?
+      ).must_equal(false)
+    end
+
+    it('returns DROP if credits_per_second is negative') do
+      sampler = OpenTelemetry::Sampling::RateLimited::Sampler.new(-1)
+
+      _(
+        sampler.should_sample?(
+          trace_id: SecureRandom.uuid.to_s,
+          parent_context: nil,
+          links: [],
+          name: SecureRandom.uuid.to_s,
+          kind: :internal,
+          attributes: {}
+        ).sampled?
+      ).must_equal(false)
+    end
+
+    it('adheres to the credits_per_second limit') do
+      sampler = OpenTelemetry::Sampling::RateLimited::Sampler.new(3)
+
+      3.times do
+        _(
+          sampler.should_sample?(
+            trace_id: SecureRandom.uuid.to_s,
+            parent_context: nil,
+            links: [],
+            name: SecureRandom.uuid.to_s,
+            kind: :internal,
+            attributes: {}
+          ).sampled?
+        ).must_equal(true)
+      end
+
+      3.times do
+        _(
+          sampler.should_sample?(
+            trace_id: SecureRandom.uuid.to_s,
+            parent_context: nil,
+            links: [],
+            name: SecureRandom.uuid.to_s,
+            kind: :internal,
+            attributes: {}
+          ).sampled?
+        ).must_equal(false)
+      end
+
+      sleep(1)
```

Comment:
Something to consider with tests that require sleeping is that other factors may result in non-deterministic behavior of tests like GC pauses. 

You may want to consider simulating time calculations using test doubles